packages/driver/src/utils.ts

```diff
@@ -65,3 +67,48 @@ export interface CryptoUtils {
   H: (msg: Uint8Array) => Promise<Uint8Array>;
   HMAC: (key: Uint8Array, msg: Uint8Array) => Promise<Uint8Array>;
 }
+
+const _tokens = new WeakMap<ResolvedConnectConfig, string>();
+
+export type FetchWrapper = (
+  path: string,
+  init: RequestInit
+) => Promise<Response>;
+
+export async function makeFetch(
```

Comment:
Nitpick: Should we name this something like `AuthenticatedFetch`/`AuthedFetch` and `makeAuthFetch`, so it's more clear what it does?

Reply:
Yeah, this is a good question. Furthermore, I wonder if `fetch` is even really the best name for this since it ends up encouraging you to shadow the `fetch` global which can be confusing.

Also, it really does slightly more than authentication, it also sets up the base URL, and maybe more things in the future. The thing it does is roughly: "Make an authenticated HTTP request a specific EdgeDB branch", so maybe there is something there? I'll think of a few names as well.

Reply:
Hm, I am not completely sure, I renamed the util func to getAuthenticatedFetch and it returns AuthenticatedFetch... it can maybe also be AuthenticatedDBFetch... but yes, not 100% what is the most future proof

Reply:
I was thinking something like `ServerRequest` or something like that? Still not sure 😅